### ISSUE TITLE
RedisServer is ready support

### ISSUE THREAD

**Issue by penev-ff** [fppt/jedis-mock#475]
Currently, the RedisServer class lacks a convenient mechanism to notify external code when the server has fully started and is ready to accept requests. I would be happy if a method or mechanism is introduced in the RedisServer class that signals when the server has completed its startup and initialization processes. 

Example.
```java

// Example if server is already running:
server.stop();
// - Here assert request that are using the mock server fails
server.start();
Awaitility.await().until(server::isReady);
// Proceed with assertions or operations requiring the server to be operational
```

**Comment by inponomarev** [comment]
1. Do I get this right that  you are getting the connection error after `server.start()` as the connection attempt is made before the port is open? 
2. What do you think if `server.start()` will simply block until the server is ready instead?

**Comment by inponomarev** [comment]
Ok I have investigated that if tests are run in the same thread, `server` is ready to accept connections after `start()`.

However, there are two corner case usages for `isReady`: 
1. If server is started in one thread, but is being connect from a separate thread.
2. If user "forgets" about the state of the server, e. g. doesn't know if `start` or `stop` has been called.

I have implemented `isRunning` method in https://github.com/fppt/jedis-mock/pull/483
